ducktape/utils/wait.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2015 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#
+# Conscious waiting
+#
+#
+#
+
+import time
+import traceback
+import os
+import errno
+import signal
+
+class TimeoutError (Exception):
+    pass
+
+time_saved_sum = 0
+
+def time_saved ():
+    """ Returns the amount of time saved using wait.until() instead of time.sleep() """
+    global time_saved_sum
+    return time_saved_sum
+
+
+def sleep (seconds):
+    """ Simple wrapper for time.sleep() but indicates that the programmer
+        made a conscious decisions not to use wait.until() """
+    return time.sleep(seconds)
+
+def until0 (timeout, expect_result, test_func, *args):
```

Comment:
Normally in Python you'd just use a `lambda` and the whole question about `*args` and `**kwargs` would be moot.

On naming, agree that `until0` is a very C-like name, though I'm not sure we want to treat this as public interface -- the two wrappers already cover all the cases, passing in the expected result in explicitly rather than using the easier-to-read wrappers should probably be discouraged. `_until` would give the indication that it's private and avoid `*` imports.